src/qibolab/instruments/rfsoc.py

```diff
@@ -512,12 +508,12 @@ def get_if_python_sweep(self, sequence: PulseSequence, qubits: List[Qubit], *swe
     def convert_sweep_results(
         self,
         original_ro: PulseSequence,
-        qubits: List[Qubit],
-        toti: List[float],
-        totq: List[float],
+        qubits: list[Qubit],
+        toti: list[float],
+        totq: list[float],
         execution_parameters: ExecutionParameters,
-    ) -> Dict[str, Union[IntegratedResults, SampleResults]]:
-        """Convert sweep res to qibolab dict res
+    ) -> dict[str, Union[IntegratedResults, SampleResults]]:
```

Comment:
About these: @stavros11 was trying to keep implicit support for py3.8, as long as you were not using Zurich.

To be fair, I'm not sure is really that relevant, since `pip` will refuse to install for py3.8 (it's written in the package metadata that is not supported).
In order to use it, you should download the repo, change `pyproject.toml` (extending Python range and removing Zurich) and install from source.

So, before keep going with all these changes, I wanted to prompt a discussion and an agreement about which Python versions we are truly supporting. If it's py3.9+, as declared, then we can incrementally adopt the new features everywhere (which mainly consist of the new type hints, since the biggest part of py3.9 was the internal rework of the parser https://docs.python.org/3.9/whatsnew/3.9.html).

Reply:
This doesn't make too much sense to me and I don't see any real advantage... but ok, let's discuss it

Reply:
There is certainly an advantage in supporting living versions of Python, and it's reasonable that we only had to give up on py3.8 because of Laboneq.

However, as I highlighted above, at this point it is already a pain to use Qibolab with py3.8, so I'd go all-in for py3.9+ myself. But I accept other opinions.

Reply:
I agree that "unofficially supporting" py3.8 doesn't make much sense, so personally I am fine with leaving the new type hints. I am not sure if anyone is using qibolab with py3.8 and for some reason cannot update to newer versions, but I believe we had this discussion before when we introduced LaboneQ and since nobody complained so far, it should be fine to drop support (officially and unofficially).